### PR TITLE
Update user

### PR DESCRIPTION
**Pull request by vladimir-kozyrev**
https://github.com/OpenVPN/terraform-provider-openvpn-cloud/issues/21
- Add update user functionality.
- Replace deprecated `ioutil.ReadAll` with `io.ReadAll`.
- Remove unused `getNetworkConnectorSlice` func.
- Fix some minor `golangci-lint` issues.
- Add linter that runs for every PR commit.
- Add [acceptance test](openvpncloud_user) for `openvpncloud_user`. It creates, reads, updates, and deletes the user using provider code.

### CODE REVIEW

File: openvpncloud/resource_network.go [hunk 1]
```diff
@@ -336,25 +336,3 @@ func resourceNetworkDelete(ctx context.Context, d *schema.ResourceData, m interf
 	}
 	return diags
 }
-
```
Comment (patoarvizu) [inline openvpncloud/resource_network.go:350]: Was this function not used? I may have copy/pasted something at some point and left it there inadvertently 🙂

Comment (vladimir-kozyrev) [reply openvpncloud/resource_network.go:350]: Yep, you can `grep` or search for it using some other means. `golangci-lint` also tells me that it's redundant.

**Review by michaelfmnk — APPROVED** [review]

File: openvpncloud/resource_user_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccOpenvpncloudUser_basic(t *testing.T) {
+	rn := "openvpncloud_user.test"
+	user := client.User{
+		Username:  acctest.RandStringFromCharSet(10, alphabet),
+		FirstName: acctest.RandStringFromCharSet(10, alphabet),
+		LastName:  acctest.RandStringFromCharSet(10, alphabet),
+		Email:     "v.kozyrev.sa@gmail.com", // replace it with a test email later
```
Comment (michaelfmnk) [inline openvpncloud/resource_user_test.go]: please, replace this before merging to main

Comment (vladimir-kozyrev) [reply openvpncloud/resource_user_test.go]: fixed

**Pull request merged**